API/TreeData/TreeDataMock.cs

```diff
@@ -0,0 +1,149 @@
+
+using MySql.Data;
+using MySql.Data.MySqlClient;
+
+namespace TreeData_CLI
+{
+    public class TreeRequest
+    {
+        public string Subquadrat { get; set; }
+        public string Tag { get; set; }
+        public string StemTag { get; set; }
+        public string SpCode { get; set; }
+        public double DBH { get; set; }
+        public string Codes { get; set; }
+        public string Comments { get; set; }
+        public bool WasDead(ArrayList Trees){
+            foreach(TreeStorage tree in Trees){
+                if(tree.Tag == Tag && tree.StemTag == StemTag
+                    && tree.Codes.Contains("dt") ){
+                    return true;
+                }
+            }
+            return false;
+        }
+    }
+    public class TreeStorage {
+        public int TempID { get; set; }
+        public string QuadratName { get; set; }
+        public string Tag { get; set; }
+        public string StemTag { get; set; }
+        public string SpCode { get; set; }
+        public double DBH { get; set; }
+        public string Codes { get; set; }
+        public string HOM { get; set; }
+        public string ExactDate { get; set; }
+        public double x { get; set; }
+        public double y { get; set; }
+        public string PlotID { get; set; }
+        public string CensusID { get; set; }
+        public string Errors { get; set; }
+
+        public TreeStorage MapSQL(MySqlDataReader rdr){
+            TempID = rdr.GetInt32("TempID");
+            QuadratName = rdr["QuadratName"].ToString();
+            Tag = rdr["Tag"].ToString();
+            StemTag = rdr["StemTag"].ToString();
+            SpCode = rdr["SpCode"].ToString();
+            DBH = rdr.GetDouble("DBH");
+            Codes = rdr["Codes"].ToString();
+            HOM = rdr["HOM"].ToString();
+            ExactDate = rdr["ExactDate"].ToString();
+            x = rdr.GetDouble("x");
+            y = rdr.GetDouble("y");
+            PlotID = rdr["PlotID"].ToString();
+            CensusID = rdr["CensusID"].ToString();
+            Errors = rdr["Errors"].ToString();
+            return this;
+        }
+    }
+    public class TreeResponse
+    {
+        public string Subquadrat { get; set; }
+        public string Tag { get; set; }
+        public string StemTag { get; set; }
+        public int ErrorCode { get; set; }
+        public string Error { get; set; }
+        public TreeResponse (TreeRequest tree, int ecode, string error){
+            Subquadrat = tree.Subquadrat;
+            Tag = tree.Tag;
+            StemTag = tree.StemTag;
+            ErrorCode = ecode;
+            Error = error;
+        }
+    }
+    public static class TreeDataMock
+    {
+        private static string connStr = System.Environment.GetEnvironmentVariable("MySQLConnection", EnvironmentVariableTarget.Process);
+        private static MySqlConnection conn = new MySqlConnection(connStr);
+
+        private static ArrayList QueryTreeDB(string sql){

```

Comment:
Let's create an abstraction for the client we use to connect to MySQL. Call it `MySqlClient` and put it in a file `WebApi/Configuration/MySqlClient.cs`.

Reply:
When you say that, do you mean an actual abstract class, or just a standard class the functions to abstract that logic? I moved the connection out to a standard class here: https://github.com/ForestGeoHack/ForestGEO/pull/12/commits/2cbd18ca48c4610744cb74306943759297f1f09a
Let me know if you were thinking of something else 😄

Reply:
Please move the file to `WebApi/Configuration/` and change the namespace accordingly, i.e. `ForestGEO.WebApi.Configuration`.

When I say abstraction, I just mean to create an object that encapsulates to logic we use to read from SQL. You did a great job of doing just that! One small suggestion is to create a constructor that takes in a connection string.

Reply:
Woo! Glad I got that. Also, moved this around in https://github.com/ForestGeoHack/ForestGEO/pull/12/commits/fe58d2c61443f9fa0bdb43b61cb010d9518a75db